### ISSUE TITLE
doesn't work properly since android update version 9

### ISSUE THREAD

**Issue by R3drumzz** [gabm/TapAndTurn#56]
Hi, I've been using this app for a year or two now and it's always worked pretty well (save the occasional required restart but that's light compared to some problems I've had with full featured apps). However it's been working pretty poorly since I updated to android 9 (it's actually oxygenOS version 9.0.0). It's always on but I don't get the prompts to switch screen rotations anymore even if I restart the app or phone.

I'm available to answer any questions you have or to test anything specific.

regards

Chris

**Comment by R3drumzz** [comment]
ok I'll have a look at that thread but I'm not in a pixel I'm on a oneplus so I'm not exactly running the same android. I'll report back after I have a look at thread 52.

**Comment by licaon-kter** [comment]
@R3drumzz it's Pie....same issue.

**Comment by tristan-k** [comment]
I'm on lineageos 16 (android 9) and the service is running but no icon is displayed.

**Comment by just-Nob** [comment]
I've the same problem since I switched to LineageOS16.0 (which is on fact Android 9.0 / Pie) on my BQ Aquaris X Pro a few days ago.

Anyway, it just works for a short period of time - let's say a half of a minute, maybe one or two minutes - after activation (this could also be triggered by deactivating and reactivating the service by tapping Tap'n'Turn's Status Bar Notification twice, for example), then the service seems to be stopped, I.e. the "rotation circle" in the phone's edges isn't displayed anymore.

**Comment by gerroon** [comment]
Same issue with LO Pie

**Comment by just-Nob** [comment]
Since on my BQ Aquaris X Pro - currently running LOS 17.1 - _Tap 'n' Turn_ runs correctly for two or three minutes before it seems to be killed, I tried to disable Android's _run time optimisation_ (which will push inactive apps out of RAM after some time to save battery power) for _Tap 'n' Turn_ in Android's settings, but unfortunately without being successful. :-(

@R3drumzz , @licaon-kter , @tristan-k , @gerroon : Is anyone of you able to confirm the behaviour described by me, I.e. _Tap 'n' Turn_ runs correctly for a short period of time everytime after being newly activated?
As I understood, most of you, but not everyone runs LOS?

Does anyone else have an idea why _Tap 'n' Turn_ may be killed, and if so, how to prevent that?

It's a pity that the original author doesn't seem to be interested in further development. :-(

**Comment by mcg-android** [comment]
> @R3drumzz , @licaon-kter , @tristan-k , @gerroon : Is anyone of you able to confirm the behaviour described by me, I.e. _Tap 'n' Turn_ runs correctly for a short period of time everytime after being newly activated? As I understood, most of you, but not everyone runs LOS?

I'm not one of them, but I just moved from one version of /e/OS (which is based on LineageOS) identifying itself as "Android 8.1.0" to a younger version of /e/OS identifying itself as "Android 12", and I see that same behaviour:

Tap 'n' Turn works perfectly for a short while (longer than a few seconds, but a few minutes at most) after each "activate", and then the icon is no longer displayed after a physical rotation of the device. Android continuously warns the user (in the form of an Android notification) that Tap 'n' Turn is displaying over other apps.

The stopping does not appear to be related to the screen turning off and/or getting locked, i.e. it's not exactly #52. It does, however, look as if Tap 'n' Turn no longer gets informed about the physical rotation.

**Comment by just-Nob** [comment]
I'm currently on Android 11 (ArrowOS), and on this, T'n'T doesn't seem to work at all.
Fortunately this isn't that hard for me on this system, because ArrowOS does have this functionality built-in, even a little smarter.

**Comment by gabm** [comment]
*original author here*

thank you for your continued interest in Tap'n'Turn. Unfortunately it has not been maintained for a very long time and i don't have the time nor the need to develop this further, so please don't expect this to be fixed anytime soon. 

That being said, I would support anyone to put effort into this, if there is interest...

**Comment by licaon-kter** [comment]
Did everyone test https://f-droid.org/packages/com.jarsilio.android.autoautorotate/ yet?

**Comment by gabm** [comment]
yes,  it's a great piece of software.. a bit different than T'n'Ts original mission, but works very well!

**Comment by just-Nob** [comment]
That's right, Auto Auto-Rotate is nice, but in my opinion, it focuses on another use case.

**Comment by RamKromberg** [comment]
show stopper for me as well.

@gabm @just-Nob honorable mention to [autogyro](https://github.com/zlsa/autogyro) which, even if it's not as convenient user-interface wise, fits the use case and works where tapandturn doesn't (various android 10 and 11 roms).

**Comment by Atemu** [comment]
Note that, in newer versions of LineageOS, disabling auto-rotation lets you rotate in a TapAndTurn manner. A small black button appears in the bottom right corner when you rotate the device.

I don't know if it's a thing in 19.1 but it is in 20.0. I also don't know whether this is a general Android thing or a LineageOS thing but it might be.

**Comment by just-Nob** [comment]
Hi @Atemu ,

thx for that information, because I thought about to switch to LOS.
Currently I'm stuck at an Android 11 based version of ArrowOS, which has the same feature, and that's one of the most important ones to me! 😇
Nice to hear that the T'n'T spirit is still alive.
